### PR TITLE
[Pallas] Make gmm support bf16

### PR DESCRIPTION
**Pull request by alanwaketan**
Summary:
This pull request does:
1. make gmm support bf16,
2. don't visit_empty_groups for gmm,
3. make the reference gmm torchy.

Test Plan:
python test/test_gmm.py

### CODE REVIEW

File: torch_xla/experimental/custom_kernel.py [hunk 1]
```diff
@@ -56,38 +56,43 @@ def jax_import_guard():
...
-      return jnp.float32
-    elif dtype == torch.float64:
-      if _XLA_USE_BF16:
-        return jnp.bfloat16
-      return jnp.float64
-    elif dtype == torch.float16:
-      return jnp.float16
-    elif dtype == torch.bfloat16:
+  if dtype == torch.float32:
+    if _XLA_USE_BF16:
```
Comment (JackCaoG) [inline torch_xla/experimental/custom_kernel.py:66]: .. we should really try to just cast everything to bf16 manually after we are done with current deadline...

**Review by JackCaoG — APPROVED** [review]

File: torch_xla/experimental/custom_kernel.py (hunk 1 above)
Comment (alanwaketan) [reply torch_xla/experimental/custom_kernel.py:66]: This is just for backward compatibility....

**Comment by alanwaketan** [comment]
Thanks Jack for approving the change.

**Comment by alanwaketan** [comment]
Skip GPU CI to move fast...

**Pull request merged**